### PR TITLE
fix: parsing url and jsonTransportMode warning 

### PR DESCRIPTION
**Pull request by adrhub**
Related issue: https://github.com/deepstreamIO/deepstream.io-client-js/issues/515

### CODE REVIEW

File: src/util/utils.ts [hunk 1]
```diff
@@ -1,10 +1,4 @@
-let UniformResourceLocator: any
-
-if (typeof URL === 'undefined') {
-  UniformResourceLocator = require('url').URL
-} else {
-  UniformResourceLocator = URL
-}
+const URL: any = require('url')
```
Comment (yasserf) [inline src/util/utils.ts]: I think the idea behind this was to keep the bundle size small by not importing the URL module via webpack

File: src/connection/socket-factory.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ const BrowserWebsocket = (global.WebSocket || global.MozWebSocket) as any
 
 export type SocketFactory = (url: string, options: JSONObject, heartBeatInterval: number) => Socket
 
-export const socketFactory: SocketFactory = (url, options = { jsonTransportMode: false }, heartBeatInterval) => {
```
Comment (yasserf) [inline src/connection/socket-factory.ts]: why was this removed?

Comment (yasserf) [reply src/connection/socket-factory.ts]: can we delete it instead from options? If someone sets it to true it would give the same error

File: src/util/utils.ts (hunk 1 above)
Comment (adrhub) [reply src/util/utils.ts]: I understand but react native doesn't have implementation for URL:
https://github.com/facebook/react-native/blob/e6057095adfdc77ccbbff1c97b1e86b06dae340b/Libraries/Blob/URL.js#L157-L159
It makes it impossible to use this library after this update

File: src/connection/socket-factory.ts (hunk 2 above)
Comment (adrhub) [reply src/connection/socket-factory.ts]: I didn't remove it to support backward compatibility in case someone uses it for some reason, do you think we can remove it ?

File: src/util/utils.ts (hunk 1 above)
Comment (yasserf) [reply src/util/utils.ts]: Does react native not have the window.URL same as browsers?

Comment (adrhub) [reply src/util/utils.ts]: Unfortunately no
